lib/hanami/app.rb

```diff
@@ -53,6 +54,32 @@ def prepare_base_load_path
         $LOAD_PATH.unshift(base_path) unless $LOAD_PATH.include?(base_path)
       end
 
+      # Uses [dotenv](https://github.com/bkeepers/dotenv) (if available) to populate `ENV` from
+      # various `.env` files.
+      #
+      # For a given `HANAMI_ENV` environment, the `.env` files are looked up in the following order:
+      #
+      # - .env.{environment}.local
+      # - .env.local (unless the environment is `test`)
+      # - .env.{environment}
+      # - .env
+      #
+      # If dotenv is unavailable, the method exits and does nothing.
+      def load_dotenv
+        hanami_env = Hanami.env
+        dotenv_files = [
+          ".env.#{hanami_env}.local",
+          (".env.local" unless hanami_env == :test),
+          ".env.#{hanami_env}",
+          ".env"
+        ].compact
+
+        require "dotenv"
+        Dotenv.load(*dotenv_files) if defined?(Dotenv)
```

Comment:
@timriley I want to double check this with you: what's the Public API to access env variables?

In H1 we had `Hanami::Env`, reasons are written here: https://lucaguidi.com/2016/12/27/isolate-global-state/

---

I'm asking because we could populate `Hanami::Env`, instead of `ENV`, by using `Dotenv.parse(*dotenv_files)`, which returns a `Hash`.

Reply:
@jodosha It's a good question. This PR certainly doesn't have the ambition of providing such an interface — its job is just to populate `ENV` as we've always done, but just in a clearer way.

As for whether we should provide a first party, "safer" interface to ENV, my current feeling is that it's probably not worthwhile it at this point: we should be encouraging our users to take advantage of our settings components to access relevant values from the env. Cases where that is not possible (like some of the scenarios I described in my comment above) are hopefully relatively minor and not the kind of thing that would be close to the actual working business logic of Hanami apps.

What do you think?